core/tools.js

```diff
@@ -1518,6 +1518,76 @@
 		},
 
 		/**
+		 * Converts hexstring to base64 coode.
+		 *
+		 * Hexstring contains `8bit` numbers, where base64 use `6bit` to code.
+		 * We need to process `3 * 8bit` chunks of hexstring into `4 * 6bit` characters of base64.
+		 *
+		 * Algorithm:
+		 *
+		 * 1. Take `3 * 8bit`.
+		 * 2. If there is less than 3 byte, fill empty bits with zeros.
+		 * 3. Transform `3 * 8bit` into `4 * 6bit` numbers.
+		 * 4. Translate those numbers to proper character related to base64.
+		 * 5. If there was added extra zeros bytes fill them with `=` sign.
+		 *
+		 * Example:
+		 *
+		 * * Hex: `08A11D8ADA2B` -> binary: `0000 1000 1010 0001 0001 1101 1000 1010 1101 1010 0010 1011`.
+		 * * `|` (pipe) shows where base64 will cut bits during transform.
+		 * * Binary: `0000 10|00 1010| 0001 00|01 1101| 1000 10|10 1101| 1010 00|10 1011`.
+		 * * Now we have 6bit numbers (decimal values below), which are translated to indexes in `base64characters` array.
+		 * * Decimal: `2 10 4 29 34 45 40 43` -> base64: `CKEditor`.
+		 *
+		 * @since 4.8.0
+		 * @param {String} hexString String contained only hexadecimal values, e.g. `0AB7`.
+		 * @returns {String} Base64 string represnet input value.
+		 */
+		convertHexstringToBase64: function( hexString ) {
```

Comment:
There's one more thing, I thought that I have included it in original review, but turns out I forgot to paste it in review summary.

Rather than adding `convertHexstringToBase64` which does the job for your particular purpose, you should do two separate function:

* `convertHexStringToBytes`
* `convertBytesToBase64`

That way this code will be more reusable, and can be used by a developer that looks only to use bytes to base64 conversion, etc.